google/cloud/aiplatform/prediction/local_model.py

```diff
@@ -438,6 +438,9 @@ def deploy_to_local_endpoint(
     ):
```

Comment:
Since LocalEndpoint implements __enter__ and __exit__ methods anyway, we don't need the @contextmanager decorator on this function. (https://docs.python.org/3/library/contextlib.html#contextlib.contextmanager) 

It should just create and return the LocalEndpoint object instead of using the try/yield/finally pattern.